File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/service/RegistrationService.java

package com.alpha.ogo.service;

import com.alpha.ogo.domain.Registration;
import com.alpha.ogo.domain.User;
import com.alpha.ogo.dto.CommonResponce;

/**
 * @author Tharindu
 *
 */
public interface RegistrationService {

	CommonResponce<Registration> createRegistration(Registration registration);

	CommonResponce<User> checkUsername(String username);

	CommonResponce<Registration> getAllUsers();

	CommonResponce<Registration> getRegistrationById(String registrationId);

}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/service/RegistrationService.kt
package com.allion.ogoss.service

import com.allion.ogoss.domain.Registration
import com.allion.ogoss.domain.User
import com.allion.ogoss.dto.CommonResponce

interface RegistrationService {

    fun createRegistration(registration: Registration?): CommonResponce<Registration>

    fun createUserRegistration(user: User?): CommonResponce<User>

    fun checkUsername(username: String?): CommonResponce<User?>?

    fun getAllUsers(): CommonResponce<Registration?>?

    fun getRegistrationById(registrationId: String?): CommonResponce<Registration?>?

}
